### PR TITLE
Article for day 14.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

**Comment by sjrd** [comment]
Capture d’écran 2021-12-14 à 17 47 10

File: docs/puzzles/day14.md [hunk 1]
```diff
@@ -1,11 +1,310 @@
...
+
+Inductive case: a pair $xy$, and $n > 0$
+
+$$
+\begin{aligned}
+S(xy, n) & = S(xzy, n-1) \text{ where $z$ is the insertion char for the pair $xy$ (by definition)} \\
+         & = S(xz, n-1) + S(zy, n-1) \text{ -- the particular case of 3 chars above}
+\end{aligned}
+$$
+
```
Comment (tgodzik) [inline docs/puzzles/day14.md:203]: Could we add something like:
`This means that occurrences of letters after xy has produced it's final polimer are equal to the occurences of xzy after 1 less step`

**Review by tgodzik — CHANGES_REQUESTED** [review]
Some minor comments, but it looks great!

File: docs/puzzles/day14.md [hunk 2]
```diff
@@ -1,11 +1,310 @@
...
+Now that we have an inductive defition of $S(xy, n)$ for all pairs, we can write an interative algorithm that computes that for all possible pairs $xy$, for $n \in \lbrack 0, 40 \rbrack$ :
```
Comment (julienrf) [inline docs/puzzles/day14.md:204]:
```suggestion
Now that we have an inductive definition of $S(xy, n)$ for all pairs, we can write an iterative algorithm that computes that for all possible pairs $xy$, for $n \in \lbrack 0, 40 \rbrack$ :
```

File: docs/puzzles/day14.md [hunk 3]
```diff
@@ -1,11 +1,340 @@
...
+<Solver puzzle="day14-part1"/>
+
+## Part 2
+
+The statement of part 2 says to do the same thing with 40 iterations.
+Of course, it is not as simple as reusing the first implementation above.
+The final polymer's length would exceed 4G of characters, let alone the additional memory of the `List` and the processing time required to build it.
+We need to compute the eventual frequencies *without* computing the `finalPolymer` at all.
+
+Out of the blue, we define the abstract function $S(ab \cdots f, n)$ as being the frequencies of all letters, *except the first one*, in the expansion of $ab \cdots f$ after $n$ steps.
```
[the remark this answers is not in the document]
Comment (sjrd) [reply docs/puzzles/day14.md:153]: That's what I intended. It's hard to explain how we come up with that function and that definition. It's intuition, that is only rationalized after the fact by the fact that the "proof" works out.

**Review by julienrf — APPROVED** [review]

File: docs/puzzles/day14.md [hunk 4]
```diff
@@ -1,11 +1,340 @@
...
+
+$$
+\begin{aligned}
+S(\text{NNCB}, 0) & = \{ \text{B} \rightarrow 1, \text{C} \rightarrow 1, \text{N} \rightarrow 1 \} \\
+S(\text{NNCB}, 1) & = S(\text{NCNBCHB}, 0) \\
+                  & = \{ \text{B} \rightarrow 2, \text{C} \rightarrow 2, \text{H} \rightarrow 1, \text{N} \rightarrow 1 \}
+\end{aligned}
+$$
+
+Note that one `N` is excluded from the frequencies every time---the one at the beginning of the string.
```
[the remark this answers is not in the document]
Comment (julienrf) [reply docs/puzzles/day14.md:164]: It’s supposed to be a dash, but it’s not correctly rendered:
Screenshot from 2021-12-14 18-05-47

[the remark this answers is not in the document]
Comment (julienrf) [reply docs/puzzles/day14.md:164]: You may want to use parens instead.
```suggestion
Note that one `N` is excluded from the frequencies every time (ie, the one at the beginning of the string).
```

**Comment by sjrd** [comment]
Thanks for the reviews. I applied all your suggestions, except the "out of the blue".

File: docs/puzzles/day14.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (tgodzik) [reply docs/puzzles/day14.md:153]: So maybe:
```suggestion
Intuitively, let's start with defining the abstract function $S(ab \cdots f, n)$ as being the frequencies of all letters, *except the first one*, in the expansion of $ab \cdots f$ after $n$ steps.
```

?

`out of the blue` doesn't sound right in this situation.

File: docs/puzzles/day14.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (sjrd) [reply docs/puzzles/day14.md]: I used parens ;)

File: docs/puzzles/day14.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sjrd) [reply docs/puzzles/day14.md]: OK, I tried a definition formulation:
```md
Let us start by defining the abstract function $S(ab \cdots f, n)$ as being the frequencies of all letters, *except the first one*, in the expansion of $ab \cdots f$ after $n$ steps.
We come up with this function definition based on mathematical intuition.
We can only justify that choice by showing hereafter that it allows us to solve our problem.
```

[the remark this answers is not in the document]
Comment (tgodzik) [reply docs/puzzles/day14.md]: Thanks!

**Review by tgodzik — APPROVED** [review]

**Pull request merged**